### PR TITLE
feat: upgrade to node LTS v16

### PR DESCRIPTION
**Pull request by gibaros**
feat(mojaloop/#2767): upgrade nodeJS version for quoting-service - https://github.com/mojaloop/project/issues/2767
- standardised CI scripts, however, the test-integration CI job needed minor tweaks to work with the quoting-service's implementation
- updated docker-compose files to work correctly
- added wait-4-docker npm script
- fixed lint issues
- updated .nvmrc to latest LTS version
- added standard CI scripts/config to package.json: release, snapshot, standard-version, etc
- updated gitignore to include test/results and IGNORE patterns
- updated README with standard auto-release information
-  Fixed imports
- Cleaned up Package.json
- Removed unnecessary dependencies
- Removed unnecessary files

Notes:
- Helm chart mountPaths need to be updated from /opt/quoting-service to /opt/app as follows:
```yaml
    volumeMounts:
    - mountPath: /opt/app/config
      name: <deployment-name>
```

BREAKING CHANGE: Major version bump for node v16 LTS support, re-structuring of project directories to align to core Mojaloop repositories and docker image now uses /opt/app instead of /opt/quoting-service which will impact config mounts.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM node:12.16.1-alpine as builder
```
Comment (mdebarros) [inline Dockerfile:1]: We also need to re-align this dockerfile.

File: package.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
   "name": "quoting-service",
```
Comment (mdebarros) [inline package.json:2]: We are missing a couple of things here...

Npm scripts `replace`, as well as the `standard-version` config overrides, and standardization of the test npm scripts.

File: src/server.js [hunk 3]
```diff
@@ -146,17 +146,17 @@ const initServer = async function (db, config) {
...
-          console: [{
-            module: 'good-squeeze',
-            name: 'Squeeze',
-            args: [{ log: '*', response: '*' }]
-          }, {
-            module: 'good-console',
-            args: [{ format: '' }]
-          }, 'stdout']
         }
+        // reporters: {
```
Comment (mdebarros) [inline src/server.js:150]: If we need to comment this out, we should add a `// TODO: ` with some information about why it's necessary. If not, then we should remove this code block completely.

**Review by mdebarros — COMMENTED** [review]
Hey @gibaros, a couple of comments for your attention.

Happy to jump on a call to discuss them with you in more detail.

Also, take note that all the checks need to pass, and we need to add a proper description to this PR --> please refer to the following as an example: https://github.com/mojaloop/central-ledger/pull/902

File: Dockerfile (hunk 1 above)
Comment (gibaros) [reply Dockerfile:1]: done

File: package.json (hunk 2 above)
Comment (gibaros) [reply package.json:2]: added, ready for final review

File: src/server.js (hunk 3 above)
Comment (gibaros) [reply src/server.js:152]: TODO was added

File: package.json [hunk 4]
```diff
@@ -32,76 +33,95 @@
...
     "rc": "1.2.8"
   },
   "devDependencies": {
-    "eslint": "7.32.0",
-    "jest": "27.5.1",
-    "jest-junit": "13.0.0",
+    "eslint": "8.16.0",
+    "jest": "28.1.1",
+    "jest-junit": "13.2.0",
     "npm-audit-resolver": "2.3.1",
```
Comment (mdebarros) [inline package.json:116]:
```suggestion
    "npm-audit-resolver": "3.0.0-7",
```

File: package.json [hunk 5]
```diff
@@ -32,76 +33,95 @@
...
     "json-rules-engine": "5.0.2",
-    "knex": "1.0.3",
+    "knex": "2.1.0",
     "memory-cache": "0.2.0",
-    "minimist": "1.2.5",
+    "minimist": "1.2.6",
     "mysql": "2.18.1",
-    "node-fetch": "3.2.1",
+    "node-fetch": "3.2.5",
+    "npm-audit-resolver": "3.0.0-7",
```
Comment (mdebarros) [inline package.json:108]: Remove this

File: package.json [hunk 6]
```diff
@@ -32,76 +33,95 @@
...
-    "@mojaloop/sdk-standard-components": "15.13.0",
-    "ajv": "8.10.0",
+    "@mojaloop/sdk-standard-components": "17.0.3",
+    "ajv": "8.11.0",
     "ajv-keywords": "5.1.0",
-    "axios": "0.26.0",
+    "axios": "0.27.2",
     "blipp": "4.0.2",
-    "eslint-config-standard": "16.0.3",
+    "eslint-config-standard": "17.0.0",
```
Comment (mdebarros) [inline package.json:97]: Moved to devDependencies

File: package.json (hunk 4 above)
Comment (gibaros) [reply package.json]: fixed on Zoom session

File: package.json [hunk 7]
```diff
@@ -2,7 +2,7 @@
   "name": "quoting-service",
   "description": "Quoting Service hosted by a scheme",
   "license": "Apache-2.0",
-  "version": "14.0.0",
+  "version": "14.0.1",
```
Comment (mdebarros) [inline package.json:5]: Don't bump the version as it will be handled automatically by the CI Release process.

**Review by mdebarros — CHANGES_REQUESTED** [review]
It looks good @gibaros, just minor comments to address.

File: package.json (hunk 7 above)
Comment (gibaros) [reply package.json]: done, ready for final review

**Review by mdebarros — APPROVED** [review]

**Pull request merged**